lib/common/client/base.rb

```diff
@@ -32,8 +35,21 @@ def perform(method, path, params, headers = nil)
       def request(method, path, params = {}, headers = {})
         raise_not_authenticated if headers.keys.include?('Token') && headers['Token'].nil?
         connection.send(method.to_sym, path, params) { |request| request.headers.update(headers) }.env
-      rescue Faraday::ClientError, Timeout::Error => e
-        raise Common::Client::Errors::ClientError, e.message
+      rescue Timeout::Error, Faraday::TimeoutError
+        log_message_to_sentry(
+          "Timeout while connecting to #{config.service_name} service", :error, extra_context: { url: config.base_path }
+        )
+        raise Common::Exceptions::GatewayTimeout
+      rescue Faraday::ParsingError => e
+        # Faraday::ParsingError is a Faraday::ClientError but should be handled by implementing service
+        raise e
```

Comment:
It's because `Faraday::ParsingError` is a `Faraday::ClientError`, so ya it's to avoid it getting swallowed by `Faraday::ClientError` or later caught by a rescue for `Error` or `Exception`. In this case it wouldn't make it up to the letters service class to be handled as a 403 (because a VAAFI header error is returning HTML not JSON).